test/acceptance/filepermissions_test.go

```diff
@@ -0,0 +1,114 @@
+package acceptance
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"testing"
+	"time"
+)
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+const (
+	agentRuntime         = 1 * time.Second
+	agentConfigLocalPath = "agent_configs/minimum_config.json"
+	agentConfigPath      = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	agentConfigCopiedDir = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d"
+	translatedTomlPath   = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml"
+)
+
+func TestFilePermissions(t *testing.T) {
+	common.CopyFile(agentConfigLocalPath, agentConfigPath)
+	err := common.StartAgent(agentConfigPath, false)
+	if err != nil {
+		log.Printf("Agent failed to start due to err=%v\n", err)
+	}
+	time.Sleep(agentRuntime)
+	common.StopAgent()
+	groupResult := createGroupResult()
+	for _, testResult := range groupResult.TestResults {
+		assert.Equal(t, testResult.Status, status.SUCCESSFUL)
+	}
+	err = common.DeleteFile(agentConfigPath)
+	if err != nil {
+		log.Printf("Failed to delete config file; err=%v\n", err)
+	}
+}
+
+func createGroupResult() status.TestGroupResult {
+	testResults := make([]status.TestResult, 3)
+	testGroupResult := status.TestGroupResult{
+		Name:        "FilePermissions",
+		TestResults: testResults,
+	}
+	testResults[0] = onlyUserCanWriteToFilepath("root", agentConfigPath)
```

Comment:
Users create the JSON configuration files and they can put them wherever they want. They specify the path to the JSON config when using the control script... So isn't it possible that `/opt/aws/amazon-cloudwatch-agent/bin/config.json` doesn't exist and instead the file is at `/tmp/adam.json`? Isn't it also acceptable that the owner is not `root`?

Reply:
Yeah that was my thinking as well, but that doesn't explain the behavior change. I'm ok changing it though

Reply:
anyways i fixed it